### ISSUE TITLE
Click-through bug in Gnome 46

### ISSUE THREAD

**Issue by BillHuang2001** [NixOS/nixpkgs#309579]
### Describe the bug
Upgrade to Gnome 46 on NixOS (unstable) after https://github.com/NixOS/nixpkgs/pull/291339 introduces a click-through issue in some maximized windows. Clicks intended for the active application are passed through to the window below.

This bug has already been reported in the upstream in
1. https://gitlab.gnome.org/GNOME/mutter/-/issues/3404
2. https://gitlab.gnome.org/GNOME/mutter/-/issues/3451

The first issue is closed in the upstream, and the fix might already be included in mutter 46.1 for NixOS (unstable). However, the second issue remains open and seems more relevant to our case, as it specifically mentions fullscreen games.

### Steps To Reproduce
1. Upgrade to Gnome 46
2. Open a full-screen game in the steam
4. Try to click on different parts of the window, some parts have the click-through issue.

### Expected behavior
The click event should be correctly passed to the top application.

### Screenshots
The upstream issue contains screenshots and screencasts.

### Additional context
Following the suggestion in the second issue, I tested applying [this patch](https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/3720) and it appears to resolve the bug.

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.8.9, NixOS, 24.05 (Uakari), 24.05pre-git`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.22.0`
 - nixpkgs: `/nix/store/6433ddv0byqar86p7a3ja2vg6grq6mga-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by meutraa** [comment]
Since this drove me crazy, and who knows when the next mutter version will land, here is a fix for now.

Download the patch for the mutter fix (https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/3720.patch)

``` nix
{
  nixpkgs = {
    overlays = [
      (final: prev: {
        gnome = prev.gnome.overrideScope (
          self: super: {
            mutter = super.mutter.overrideAttrs (oldAttrs: rec {
              patches = (oldAttrs.patches or [ ]) ++ [ ./3720.patch ];
            });
          }
        );
      })
    ];
  };
}
```

**Comment by meutraa** [comment]
This has been fixed in [mutter 46.2](https://gitlab.gnome.org/GNOME/mutter/-/compare/46.1...46.2) which is now in [24.05](https://nixpk.gs/pr-tracker.html?pr=315267).
